File: tcs_manager_frontend/src/service/TestElementService.js
import axios from "axios";

const TEST_ELEMENT_BASE_API_URL = "http://localhost:8080/api/v1/test-elements/";

class TestElementService {

    getAllTestElements() {
        return axios.get(TEST_ELEMENT_BASE_API_URL)
            .then(response => response.data)
    }

    getTestElement(testElementId) {
        return axios.get(TEST_ELEMENT_BASE_API_URL + `${testElementId}`)
            .then(response => response.data)
    }

    getTestElementsByTestCaseId(testCaseId) {
        return axios.get(TEST_ELEMENT_BASE_API_URL + `tc/${testCaseId}`)
            .then(response => response.data)
    }

    createTestElement(testCaseId, testElement) {
        return axios.post(TEST_ELEMENT_BASE_API_URL + `tc/${testCaseId}`, testElement)
            .then(response => response.data)
    }

    updateTestElement(testElementId) {
        return axios.put(TEST_ELEMENT_BASE_API_URL + `${testElementId}`)
            .then(response => response.data)
    }

    partialUpdateTestElement(testElementId, testElement) {
        return axios.patch(TEST_ELEMENT_BASE_API_URL + `${testElementId}`, testElement)
            .then(response => response.data)
    }

    deleteTestElement(testElementId) {
        return axios.delete(TEST_ELEMENT_BASE_API_URL + `${testElementId}`)
            .then(response => response.data)
    }
}

export default new TestElementService();

File: tcs_manager_system/src/main/java/com/tcsManager/tcsmanager/service/TestElementService.java
package com.tcsManager.tcsmanager.service;

import java.util.List;
import java.util.Map;

import com.tcsManager.tcsmanager.entity.TestElement;
import com.tcsManager.tcsmanager.exception.TestCaseNotFoundException;
import com.tcsManager.tcsmanager.exception.TestElementNotFoundException;

import org.springframework.http.ResponseEntity;

/**
 * TestElementService
 */
public interface TestElementService {

    ResponseEntity<TestElement> getTestElement(long testElementId) throws TestElementNotFoundException;

    ResponseEntity<TestElement> createTestElement(TestElement testElement, long testCaseId)
            throws TestCaseNotFoundException;

    ResponseEntity<List<TestElement>> getAllTestElements();

    ResponseEntity<List<TestElement>> getTestElementsByTestCaseId(long testCaseId) throws TestCaseNotFoundException;

    ResponseEntity<TestElement> updateTestElement(long testElementId, TestElement testElementUpdated)
            throws TestElementNotFoundException;

    ResponseEntity<TestElement> partialUpdateTestElement(long testElementId, Map<Object, Object> fields)
            throws TestElementNotFoundException;

    ResponseEntity<?> deleteTestElement(long testElementId) throws TestElementNotFoundException;
}
